packages/ra-ui-materialui/src/list/List.spec.tsx

```diff
@@ -207,4 +207,28 @@ describe('<List />', () => {
             ).toHaveLength(1);
         });
     });
+
+    it('should render a list page with an error message when there is an error', async () => {
+        jest.spyOn(console, 'error').mockImplementation(() => {});
+        const Datagrid = () => <div>datagrid</div>;
+        const dataProvider = {
+            getList: jest.fn(() =>
+                Promise.reject({ error: { key: 'error.unknown' } })
+            ),
+        } as any;
+        render(
+            <CoreAdminContext dataProvider={dataProvider}>
+                <ThemeProvider theme={theme}>
+                    <List resource="posts">
+                        <Datagrid />
+                    </List>
+                </ThemeProvider>
+            </CoreAdminContext>
+        );
+        await waitFor(() => {
+            expect(
+                screen.getByText('ra.notification.data_provider_error')
+            ).not.toBeNull();
```

Comment:
You should either have

```suggestion
            expect(
                screen.queryByText('ra.notification.data_provider_error')
            ).not.toBeNull();
```

Or simply

```tsx
            expect(
                screen.getByText('ra.notification.data_provider_error')
            );
```

(which I prefer)

But having both `getByText` and `not.toBeNull()` is redundant